Shields.DataStructures.Async/AsyncQueue.cs

```diff
@@ -144,28 +143,27 @@ public void Enqueue(T value)
                 }
                 else
                 {
-                    dequeueQueue.Dequeue(value).Dispose();
+                    dequeueQueue.Dequeue(value);
                 }
             }
         }
 
         /// <summary>
-        /// Returns a disposable that completes all waiting DequeueAsync calls.
+        /// Completes all waiting DequeueAsync calls.
         /// </summary>
         /// <param name="value">The value to return to the waiting DequeueAsync callers.</param>
-        /// <returns>The disposable that completes all waiting DequeueAsync calls.</returns>
-        public IDisposable CompleteAllDequeue(T value)
+        public void CompleteAllDequeue(T value)
         {
-            return dequeueQueue.DequeueAll(value);
+            dequeueQueue.DequeueAll(value);
         }
         
         /// <summary>
-        /// Returns a disposable that cancels all waiting DequeueAsync calls.
+        /// Cancels all waiting DequeueAsync calls.
         /// </summary>
-        /// <returns>The disposable that cancels all waiting DequeueAsync calls.</returns>
-        public IDisposable CancelAllDequeue()
+        /// <param name="cancellationToken"></param>
```

Comment:
Please change the parameter comment to the following.

    /// <param name="cancellationToken">The cancellation token.</param>